cmd/bm-client/handlers/create_organisation.go

```diff
@@ -1,9 +1,107 @@
 package handlers
 
 import (
-	"github.com/bitmaelum/bitmaelum-suite/cmd/bm-client/pkg/vault"
+	"fmt"
+	"github.com/bitmaelum/bitmaelum-suite/internal"
+	"github.com/bitmaelum/bitmaelum-suite/internal/config"
+	"github.com/bitmaelum/bitmaelum-suite/internal/container"
+	"github.com/bitmaelum/bitmaelum-suite/internal/organisation"
+	"github.com/bitmaelum/bitmaelum-suite/internal/vault"
+	"github.com/bitmaelum/bitmaelum-suite/pkg/address"
+	"github.com/bitmaelum/bitmaelum-suite/pkg/bmcrypto"
+	pow "github.com/bitmaelum/bitmaelum-suite/pkg/proofofwork"
+	"os"
 )
 
-// CreateOrganisation will create a new organisation
-func CreateOrganisation(vault *vault.Vault, name, routing string) {
+// CreateOrganisation creates a new organisation locally in the vault and pushes the public key to the resolver
+func CreateOrganisation(vault *vault.Vault, orgName, fullName string, orgValidations []string) {
+	fmt.Printf("* Verifying if organisation name is valid: ")
+	orgAddr, err := address.NewOrgHash(orgName)
+	if err != nil {
+		fmt.Printf("not a valid organisation")
+		fmt.Println("")
+		os.Exit(1)
+	}
+	fmt.Printf("ok\n")
+
+	fmt.Printf("* Checking if your validations are correct: ")
+	val, err := organisation.NewValidationTypeFromStringArray(orgValidations)
+	if err != nil {
+		fmt.Print("\n  X it seems that one of your validations is wrong: ", err)
+		fmt.Println("")
+		os.Exit(1)
+	}
+	fmt.Printf("ok.\n")
+
+	fmt.Printf("* Checking if organisation is already known in the resolver service: ")
+	ks := container.GetResolveService()
+	_, err = ks.ResolveOrganisation(*orgAddr)
+	if err == nil {
+		fmt.Printf("\n  X it seems that this organisation is already in use. Please specify another organisation.")
+		fmt.Println("")
+		os.Exit(1)
+	}
+	fmt.Printf("not found. This is a good thing.\n")
+
+	fmt.Printf("* Checking if the organisation is already present in the vault: ")
+	var info *internal.OrganisationInfo
+	if vault.HasOrganisation(*orgAddr) {
+		fmt.Printf("\n  X organisation already present in the vault. Strange, but let's continue...\n")
```

Comment:
If organisation is already in the vault we should return an error and tell the user how to edit/change organisation settings.